dali/operators/decoder/audio/audio_decoder.h

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_DECODER_AUDIO_AUDIO_DECODER_H_
+#define DALI_OPERATORS_DECODER_AUDIO_AUDIO_DECODER_H_
+
+#include <memory>
+#include "dali/core/span.h"
+
+namespace dali {
+
+struct AudioMetadata {
+  int length;
+  int sample_rate;  /// [Hz]
+  int channels;
+  bool channels_interleaved;
+};
+
+class AudioDecoderBase {
+ public:
+  AudioMetadata Open(span<const char> encoded) {
+    Close();
+    auto ret = OpenImpl(encoded);
+    return ret;
+  }
+
+
+  void Close() {
+    CloseImpl();
```

Comment:
It doesn't add anything except more lines of code. Actually, the reverse pattern is often beneficial due to performance reasons.

Reply:
Right now it doesn't. In my opinion, however, thinking in future, AudioDecoderBase might want to make some additional steps whilst Closing, e.g. check if the file is actually opened (right now we can close not opened file, but I guess it's not that obvious, that it can be done always)